Description: set JVM max memory to 1024m.
Author: Matthias Klose <doko@ubuntu.com>
Acked-By: Anton Gladky <gladk@debian.org>
Last-Update: 2016-02-12

Index: VTK-6.3.0/Wrapping/Java/CMakeLists.txt
===================================================================
--- VTK-6.3.0.orig/Wrapping/Java/CMakeLists.txt
+++ VTK-6.3.0/Wrapping/Java/CMakeLists.txt
@@ -80,6 +80,8 @@ if(APPLE)
   set(JAVAC_OPTIONS -J-Xmx512m)
 endif()
 
+set(JAVAC_OPTIONS -J-Xmx1024m)
+
 get_property(java_modules GLOBAL PROPERTY VTK_JAVA_WRAPPED)
 foreach(module IN LISTS java_modules)
   if(NOT ${module}_EXCLUDE_FROM_WRAPPING)
