Description: remove internal vtkmpi4py dependency
Author: Nico Schlömer <nico.schloemer@gmail.com>
Last-Update: 2019-06-02

diff --git a/Parallel/MPI4Py/module.cmake b/Parallel/MPI4Py/module.cmake
index 07d789b2..26615b48 100644
--- a/Parallel/MPI4Py/module.cmake
+++ b/Parallel/MPI4Py/module.cmake
@@ -5,7 +5,6 @@ if (VTK_WRAP_PYTHON)
     DEPENDS
       vtkParallelMPI
     COMPILE_DEPENDS
-      vtkmpi4py
       vtkPython
     OPTIONAL_PYTHON_LINK
     EXCLUDE_FROM_JAVA_WRAPPING
